src/slate/context.js

```diff
@@ -68,14 +60,24 @@ export default class CounterContext extends React.Component {
     }
 
     render() {
-        return this.props.children
+        const contextValue = {
+            counters: this.props.counters,
+            counterUpdater: this,
+        }
+
+        return React.createElement(
+            CounterContext.Provider,
+            { value: contextValue, children: this.props.children },
+        )
     }
 }
 
-CounterContext.childContextTypes = {
-    counters: PropTypes.instanceOf(Immutable.Map),
-    counterUpdater: PropTypes.instanceOf(CounterContext),
-}
+export const CounterContext = React.createContext({
+    counters: new Immutable.Map(),
+    counterUpdater: new CountersProvider(),
```

Comment:
I don't think manually creating React components is supported, and I'm not comfortable with it.